ctru-rs/src/services/cfgu.rs

```diff
@@ -0,0 +1,227 @@
+//! Configuration service
+//!
+//! This module contains basic methods to retrieve and change configuration from the console.
+
+#[derive(Debug)]
+#[repr(u32)]
+pub enum Region {
+    Japan       = ctru_sys::CFG_REGION_JPN,
+    Usa         = ctru_sys::CFG_REGION_USA,
+    Europe      = ctru_sys::CFG_REGION_EUR,
+    Australia   = ctru_sys::CFG_REGION_AUS,
+    China       = ctru_sys::CFG_REGION_CHN,
+    Korea       = ctru_sys::CFG_REGION_KOR,
+    Taiwan      = ctru_sys::CFG_REGION_TWN,
+}
+
+#[derive(Debug)]
+#[repr(u32)]
+pub enum Language {
+    Japan       = ctru_sys::CFG_LANGUAGE_JP,
+    English     = ctru_sys::CFG_LANGUAGE_EN,
+    French      = ctru_sys::CFG_LANGUAGE_FR,
+    German      = ctru_sys::CFG_LANGUAGE_DE,
+    Italian     = ctru_sys::CFG_LANGUAGE_IT,
+    Spanish     = ctru_sys::CFG_LANGUAGE_ES,
+    SimpChinese = ctru_sys::CFG_LANGUAGE_ZH,
+    Korean      = ctru_sys::CFG_LANGUAGE_KO,
+    Dutch       = ctru_sys::CFG_LANGUAGE_NL,
+    Portuguese  = ctru_sys::CFG_LANGUAGE_PT,
+    Russian     = ctru_sys::CFG_LANGUAGE_RU,
+    TradChinese = ctru_sys::CFG_LANGUAGE_TW,
+}
+
+#[derive(Debug)]
+#[repr(u32)]
+pub enum SystemModel {
+    Model3DS        = ctru_sys::CFG_MODEL_3DS,
+    Model3DSXL      = ctru_sys::CFG_MODEL_3DSXL,
+    ModelNew3DS     = ctru_sys::CFG_MODEL_N3DS,
+    Model2DS        = ctru_sys::CFG_MODEL_2DS,
+    ModelNew3DSXL   = ctru_sys::CFG_MODEL_N3DSXL,
+    ModelNew2DSXL   = ctru_sys::CFG_MODEL_N2DSXL,
+}
+
+/// Represents the configuration service. No actions can be performed
+/// until an instance of this struct is created.
+///
+/// The service exits when all instances of this struct go out of scope.
+pub struct Cfgu(());
+
+impl Cfgu {
+    /// Initializes the CFGU service.
+    ///
+    /// # Errors
+    ///
+    /// This function will return Err if there was an error initializing the
+    /// CFGU service.
+    ///
+    /// ctrulib services are reference counted, so this function may be called
+    /// as many times as desired and the service will not exit until all
+    /// instances of Fs drop out of scope.
+    pub fn init() -> crate::Result<Cfgu> {
+        unsafe {
+            let r = ctru_sys::cfguInit();
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(Cfgu(()))
+            }
+        }
+    }
+
+    /// Gets system region from secure info
+    pub fn get_region(&self) -> crate::Result<Region> {
+        let mut region: u8 = 0;
+        let region_pointer: *mut u8 = &mut region;
+
+        unsafe {
+            let r = ctru_sys::CFGU_SecureInfoGetRegion(region_pointer);
```

Comment:
nit: This works too, and is a little less complicated:
```suggestion
        let mut region: u8 = 0;

        unsafe {
            let r = ctru_sys::CFGU_SecureInfoGetRegion(&mut region);
```